Better story for EXPORT_Pangolin=Off

I think it's worth to open a separate issue for this. I previously commented about this: https://github.com/stevenlovegrove/Pangolin/issues/326#issuecomment-392140263 and https://github.com/stevenlovegrove/Pangolin/issues/286#issuecomment-392143818.

If you build pangolin without the library export to the package registry at `~/.cmake/packages/...` with

```
cmake .. -DEXPORT_Pangolin=Off
```

then the examples and tests don't build 

```
CMake Error at test/log/CMakeLists.txt:2 (find_package):
  By not providing "FindPangolin.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Pangolin",
  but CMake did not find one.

  Could not find a package configuration file provided by "Pangolin"
  (requested version 0.4) with any of the following names:

    PangolinConfig.cmake
    pangolin-config.cmake

  Add the installation prefix of "Pangolin" to CMAKE_PREFIX_PATH or set
  "Pangolin_DIR" to a directory containing one of the above files.  If
  "Pangolin" provides a separate development package or SDK, be sure it has
  been installed.
```

You have to explicitly and also may not enable tests
```
cmake .. -DEXPORT_Pangolin=Off -DBUILD_TESTS=Off -DBUILD_EXAMPLES=Off
```

Maybe it would be good if cmake would check `EXPORT_Pangolin` before trying to compile tests and example. But probably it would be better if the tests and examples would not rely on the global package registry, but make sure to always find the pangolin lib from the same build tree. The current approach seems error prone if you have multiple copies of pangolin in different projects compiled at the same time, since it might pick up the wrong one.

For example, this might be caused by this (only speculation): #307

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,6 +94,7 @@ add_subdirectory("external")
 add_subdirectory("src")
 
 if(BUILD_TESTS)
+  set(Pangolin_DIR ${Pangolin_BINARY_DIR}/src)
   add_subdirectory("test")
 endif()
 

